### PR TITLE
Fix performance: delete activation from nav bar, user profile, admin overview page

### PR DESCRIPTION
**Pull request by mrnugget**
This removes this thing

screenshot_2022-10-28_12 03 20@2x

from

- global nav bar
- users profile page
- admin analytics overview page

## Why?

Customer reported severe impact on database performance after upgrading from 3.43.2 to 4.0.1. The reason for the performance regression is a change made in #40674: the `user { usageStats }` GraphQL query would now hit PostgreSQL instead of Redis to get the usage stats.

Here is the code path in 3.43.2 for `user { usageStats }`: https://github.com/sourcegraph/sourcegraph/blob/fd7599b9cbf3e80334a6167763b3c74ee9cddf9c/cmd/frontend/internal/usagestatsdeprecated/usage_stats.go#L42-L43

Here it is in 4.0+ https://github.com/sourcegraph/sourcegraph/blob/41e19f207038b4822dc9ebb052eb857e3dae9103/internal/usagestats/usage_stats.go#L110-L136

The SQL queries run by the latter have bad performance because they do multiple queries over one of the biggest tables we have: `event_logs`. Some of the performance problems with that have been addressed in https://github.com/sourcegraph/sourcegraph/pull/43564 and https://github.com/sourcegraph/sourcegraph/pull/43582.

Big problem: **these queries are run on every page load, due to the `withActivation` wrapper we had around the `Layout` component.** That wrapper would load the `usageStatistics` for the currently-logged-in user on every page load to display that onboarding checklist at the top.

On the customer's instance this resulted in slow queries piling up in the database, pinning CPU usage at 100% and exhausting the database connection limits.

After discussing the issue in Slack, [we decided to remove this component altogether](https://sourcegraph.slack.com/archives/C03TGHH1S3V/p1666951569092379?thread_ts=1666747586.140109&cid=C03TGHH1S3V), since we deem the risk low: the component has been unchanged in 3 years, it's unloved and unowned, it has performance implications, and it most likely doesn't have any effect on user onboarding.

## Test plan

- Existing tests

## App preview:

- [Web](https://sg-web-mrn-pj-remove-activation.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Review by kopancek — APPROVED** [review]
LGTM :godmode:

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| -0.78% (-22.63 kb) 🔽 | -0.12% (-17.07 kb) 🔽 | 0.05% (+5.56 kb) | 0.28% (+2) 🔺 |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/mrn+pj/remove-activation/compare-report.html;tab=live_object) for a full comparison between the commits 946c062 and 41e19f2 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/site-admin/analytics/AnalyticsOverviewPage/index.tsx [hunk 1]
```diff
@@ -24,15 +22,12 @@ import { Sidebar } from './Sidebar'
...
 
-const GET_STARTED_ALERT_KEY = 'get_started_admin_analytics_overview'
-
-interface IProps extends ActivationProps {
+interface IProps {
     history: H.History
 }
 
-export const AnalyticsOverviewPage: React.FunctionComponent<IProps> = ({ activation, history }) => {
+export const AnalyticsOverviewPage: React.FunctionComponent<IProps> = ({ history }) => {
```
Comment (umpox) [inline client/web/src/site-admin/analytics/AnalyticsOverviewPage/index.tsx:29]: I think this is fine to remove given the impact this is having, but just tagging @sourcegraph/iam for visibility. We might want to consider an alternative UI if this has more importance for site admins. (Non blocking)

**Review by umpox — APPROVED** [review]
LGTM

**Review by malomarrec — APPROVED** [review]
Great!
We (growth team), when we have capacity, plan to reinstate this in a better (and not buggy) form.
Thanks for removing the current version!

File: client/web/src/site-admin/analytics/AnalyticsOverviewPage/index.tsx (hunk 1 above)
Comment (mrnugget) [reply client/web/src/site-admin/analytics/AnalyticsOverviewPage/index.tsx:29]: @kopancek already approved and @pjlast was on the Zoom call with me when I removed it :)

We should tag @erzhtor @thenamankumar and @rrhyne here so they're aware. @malomarrec also gave his approval below, so we're good to go.

**Comment by mrnugget** [comment]
Build green. Running `main-dry-run` now.

**Review by eseliger — COMMENTED** [review]
LGTM, thanks for doing that so quickly!

**Pull request merged**